core/deployment/deployment_processor_test.go

```diff
@@ -75,19 +76,23 @@ func TestNewDeploymentProcessorTesting(t *testing.T) {
 	cli := test.NewFakeClientBuilder().AddK8sObjects(runtimeDeployment, configmap, supportingServiceList).Build()
 
 	context := operator.Context{
-		Client: cli,
-		Log:    test.TestLogger,
-		Scheme: meta.GetRegisteredSchema(),
+		Client:  cli,
+		Log:     test.TestLogger,
+		Scheme:  meta.GetRegisteredSchema(),
+		Version: "1.0-SNAPSHOT",
 	}
-	RuntimeHandler := app.NewKogitoRuntimeHandler(context)
-	Supportingservicehandler := app.NewKogitoSupportingServiceHandler(context)
+	runtimeHandler := app.NewKogitoRuntimeHandler(context)
+	supportingservicehandler := app.NewKogitoSupportingServiceHandler(context)
 
-	deploymentProcessor := NewDeploymentProcessor(context, runtimeDeployment, RuntimeHandler, Supportingservicehandler)
+	deploymentProcessor := NewDeploymentProcessor(context, runtimeDeployment, runtimeHandler, supportingservicehandler)
 	err := deploymentProcessor.Process()
 
 	assert.NoError(t, err)
 	assert.Equal(t, runtimeDeployment.Spec.Template.Spec.Containers[0].EnvFrom[0].ConfigMapRef.Name, configmap.Name)
 
+	v := make(map[string]string)
+	v[framework.KogitoOperatorVersionAnnotation] = "1.0-SNAPSHOT"
+	assert.Equal(t, v, runtimeDeployment.Annotations)
```

Comment:
could you also check `runtimeDeployment.Template.Annotations` ?

Reply:
done, also added for the supporting services.